app/controllers/api/v1/exercises_controller.rb

```diff
@@ -132,6 +132,29 @@ def update
       standard_update(@exercise, nil, user: current_api_user)
     end
 
+
+
+    ##################################################################
+    ## post save_image                                              ##
+    ##################################################################
+
+    api :POST, '/exercises/:uid/save_image', 'Save an image onto an exercise'
+    description <<-EOS
+      Saves an image asset as an attachment on an Exercise.
+
+      Unlike other API calls, this is accomplished via a multi-part form upload
+      with a file part, not as a traditional POST of JSON data
+
+      Requires a single form parameter named "image"
+
+      #{json_schema(Api::V1::AttachmentRepresenter, include: :readable)}
+    EOS
+    def save_image
+      attachment = AttachFile.call(@exercise, params[:image].tempfile).outputs[:attachment]
```

Comment:
True. It's probably enough to check that you can edit the Exercise. Here's the standard way we do this: https://github.com/openstax/openstax_api/blob/master/lib/openstax/api/roar.rb#L47

`OSU::AccessPolicy.require_action_allowed!(:update, current_api_user, model)`